objectRecognitionSource/src/test/java/io/vantiq/extsrc/objectRecognition/neuralNet/TestYoloQueries.java

```diff
@@ -0,0 +1,895 @@
+        
+        params.put("imageDate", imageDate);
+        
+        querySource(params);
+        
+        File d = new File(OUTPUT_DIR);
+        File[] dList = d.listFiles();
+        
+        assert dList.length == 6;
+        
+        for (File imageFile: dList) {
+            if (imageFile.getName().equals(IMAGE_1.get("filename")) || imageFile.getName().equals(IMAGE_2.get("filename"))) {
+                fail("Image should have been deleted locally.");
+            }
+        }
+    }
+    
+    @Test
+    public void testImageDateDeleteAfter() throws IOException {
+        // Only run test with intended vantiq availability
+        assumeTrue(testAuthToken != null && testVantiqServer != null);
+        
+        addLocalTestImages();
+        
+        Map<String,Object> params = new LinkedHashMap<String,Object>();
+        params.put("operation", "delete");
+        
+        List<String> imageDate = new ArrayList<String>();
+        imageDate.add(START_DATE);
+        imageDate.add("-");
+        
+        params.put("imageDate", imageDate);
+        
+        querySource(params);
+        
+        File d = new File(OUTPUT_DIR);
+        File[] dList = d.listFiles();
+        
+        assert dList.length == 2;
+        for (File file : dList) {
+            assert file.getName().equals(IMAGE_1.get("date") + ".jpg") || file.getName().equals(IMAGE_8.get("name") + ".jpg");
+        }
+    }
+    
+    @Test
+    public void testImageDateDeleteRange() throws IOException {
+        // Only run test with intended vantiq availability
+        assumeTrue(testAuthToken != null && testVantiqServer != null);
+        
+        addLocalTestImages();
+        
+        Map<String,Object> params = new LinkedHashMap<String,Object>();
+        params.put("operation", "delete");
+        
+        List<String> imageDate = new ArrayList<String>();
+        imageDate.add(START_DATE);
+        imageDate.add(END_DATE);
+        
+        params.put("imageDate", imageDate);
+        
+        querySource(params);
+
+        File d = new File(OUTPUT_DIR);
+        File[] dList = d.listFiles();
+        
+        assert dList.length == 5;
+        
+        for (File imageFile: dList) {
+            if (imageFile.getName().equals(IMAGE_2.get("date") + ".jpg") && imageFile.getName().equals(IMAGE_3.get("date") + ".jpg")
+                    && imageFile.getName().equals(IMAGE_4.get("date") + ".jpg")) {
+                fail("Images should have been deleted locally.");
+            }
+        }
+    }
+    
+    // ================================================= Helper functions =================================================
+    
+    public static void setupSource(Map<String,Object> sourceDef) {
+        VantiqResponse insertResponse = vantiq.insert("system.sources", sourceDef);
+        if (insertResponse.isSuccess()) {
+            core = new ObjectRecognitionCore(SOURCE_NAME, testAuthToken, testVantiqServer, MODEL_DIRECTORY);;
```

Comment:
As noted, it may make sense to have this NOT be a constant, but have it read in from gradle props/command line or something.